### PR TITLE
[MM-53339] import: dont miss out reactions from replies

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Now we import the reactions for replies. I also refactored the `TestImportimportMultiplePostLines` function into independent functions so that we can have a level of maintainability and readability in the code. It's now much easier to add new tests.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53339

#### Release Note

```release-note
Fix an issue where imported replies were missing their reactions
```

### CODE REVIEW

File: server/channels/app/import_functions.go [hunk 1]
```diff
@@ -1447,12 +1463,28 @@ func (a *App) importReplies(rctx request.CTX, data []imports.ReplyImportData, po
...
 		return model.NewAppError("importReplies", "app.post.overwrite.app_error", nil, "", http.StatusInternalServerError).Wrap(nErr)
 	}
 
+	for _, postAndReactions := range reactionsForCreateMap {
+		for _, reaction := range *postAndReactions.reactions {
+			if err := a.importReaction(&reaction, postAndReactions.post); err != nil {
+				return err
+			}
+		}
+	}
```
Comment (cpoile) [inline server/channels/app/import_functions.go:1486]: If we're going this one reaction at a time, is there areason not to do it above (line 1472)?

File: server/channels/app/import_functions.go [hunk 2]
```diff
@@ -1447,12 +1463,28 @@ func (a *App) importReplies(rctx request.CTX, data []imports.ReplyImportData, po
 				return model.NewAppError("importReplies", "app.post.save.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 			}
 		}
+		for _, created := range postsCreated {
+			reactions, ok := interimReactionsMap[created.CreateAt]
+			if !ok || reactions == nil {
```
Comment (cpoile) [inline server/channels/app/import_functions.go:1468]: Would reactions ever be nil if `ok == true` ?

File: server/channels/app/import_functions_test.go [hunk 3]
```diff
@@ -2013,579 +2013,749 @@ func TestImportimportMultiplePostLines(t *testing.T) {
...
+					Team:     &teamName,
+					Channel:  &channelName,
+					User:     &username,
+					Message:  model.NewPointer("Message with reaction"),
+					CreateAt: &reactionPostTime,
+					Reactions: &[]imports.ReactionImportData{{
+						User:      &user2.Username,
+						EmojiName: model.NewPointer("+1"),
+						CreateAt:  &reactionTime,
+					}},
```
Comment (cpoile) [inline server/channels/app/import_functions_test.go:2348]: Do we have a case where we're importing multiple different reactions and/or multiple of the same kind of reaction? (just wondering if that's something we'd want to test?)

**Review by cpoile — COMMENTED** [review]
Awesome! Just a couple little questions, nothing too serious. Thanks @isacikgoz!

File: server/channels/app/import_functions.go (hunk 1 above)
Comment (isacikgoz) [reply server/channels/app/import_functions.go:1486]: The only reason is to do so was to finish post importing logic and keeping the logic in a single place to import reactions. On L:1472 we could also import for new posts but then we'd need to maintain same logic in two places. Which didn't seem to be a necessity to me.

File: server/channels/app/import_functions.go (hunk 2 above)
Comment (isacikgoz) [reply server/channels/app/import_functions.go:1468]: It shouldn't but this is just a defensive programming :p

File: server/channels/app/import_functions_test.go (hunk 3 above)
Comment (isacikgoz) [reply server/channels/app/import_functions_test.go:2348]: That's definitely a valid case, let me add another reaction to that slice.

File: server/channels/app/import_functions.go [hunk 4]
```diff
@@ -1428,14 +1435,23 @@ func (a *App) importReplies(rctx request.CTX, data []imports.ReplyImportData, po
 
 		if reply.Id == "" {
 			postsForCreateList = append(postsForCreateList, reply)
+			if replyData.Reactions != nil && len(*replyData.Reactions) > 0 {
+				// although createAt is not unique, I think it is safe to
+				// assume that it could be near-unique especially for the same thread.
+				interimReactionsMap[reply.CreateAt] = replyData.Reactions
```
Comment (streamer45) [inline server/channels/app/import_functions.go:1441]: Could you explain what would happen if this assumption didn't hold?

**Review by streamer45 — APPROVED** [review]
Great work, especially with tests improvements.

File: server/channels/app/import_functions.go (hunk 4 above)
Comment (isacikgoz) [reply server/channels/app/import_functions.go]: Added to the comment itself 👍

**Review by DHaussermann — APPROVED** [review]
Teated and passed
- Confirmed replies keep their emoji reactions on import
- As this issue was import related and not data in the export file, I made sure export that predate this branch also have reactions when imported
- As a precaution, briefly ensured nothing seems off with Threads data 

LGTM!

Thanks @isacikgoz please merge.

**Pull request merged**